sys/crypto/modes/ccm.c

```diff
@@ -305,3 +316,71 @@ int cipher_decrypt_ccm(cipher_t *cipher,
 
     return plain_len;
 }
+
+int cipher_decrypt_ccm(cipher_t *cipher,
+                       const uint8_t *auth_data, uint32_t auth_data_len,
+                       uint8_t mac_length, uint8_t length_encoding,
+                       const uint8_t *nonce, size_t nonce_len,
+                       const uint8_t *input, size_t input_len,
+                       uint8_t *plain)
+{
+    if(_validate_mac_length_ccm(mac_length)){
+        return CCM_ERR_INVALID_MAC_LENGTH;
+    }
+    else {
+        return _cipher_decrypt_ccm(cipher, auth_data, auth_data_len, mac_length,
+                                   length_encoding, nonce, nonce_len, input,
+                                   input_len, plain);
+    }
+}
+
+int cipher_decrypt_ccms(cipher_t *cipher,
+                        const uint8_t *auth_data, uint32_t auth_data_len,
+                        uint8_t mac_length, uint8_t length_encoding,
+                        const uint8_t *nonce, size_t nonce_len,
+                        const uint8_t *input, size_t input_len,
+                        uint8_t *plain)
+{
+    if(_validate_mac_length_ccms(mac_length)){
```

Comment:
Just to double check, this is the only difference to `cipher_decrypt_ccm` above?

Reply:
Yes and no. CCM* specifies as well that:

> The nonce N shall encode the potential values for M such that one can uniquely determine from N the actually used value of M. The exact format of the nonce N is outside the scope of this specification and shall be determined and fixed by the actual implementation environment of the CCM* mode.

This part I did not address, this PR intended only to allow messages requiring only encryption. How to generate that nonce to encode M, and how to decode M from the nonce, was not required for me currently. I would address it latter if needed. But I realize this is not clear in the documentation I will add a comment.

Reply:
Did you check how this is implemented elsewhere? OpenWSN?